packages/app/hooks/use-add-wallet.ts

```diff
@@ -1,155 +1,65 @@
-      };
-    }
-  );
-
-  const wallets = user?.data.profile.wallet_addresses_excluding_email_v2;
-
-  const addWalletToBackend = async (address: string) => {
-    await axios({
-      url: "/v1/addwallet",
-      method: "POST",
-      data: { address },
-    });
-  };
-
-  const disconnectMagic = async () => {
-    const isMagicActive = await magic.user.isLoggedIn();
-
-    if (isMagicActive) {
-      await magic.user.logout();
-      setWeb3(undefined);
-    }
-  };
+  const user = useUser();
+  const wallet = useWallet();
 
-  const checkNewAddress = (address: string) => {
-    return !wallets?.find(
-      (addedWallet) =>
-        addedWallet.address.toLowerCase() === address.toLowerCase()
-    );
-  };
+  const hasNoPrimaryWallet = user?.user?.data.profile.primary_wallet === null;
 
   const addWallet = async () => {
     try {
-      let toastMessage = "";
-      const connectionStatus = state.status;
-      const shouldKillSession =
-        walletConnector.connected &&
-        (connectionStatus === "connected" || connectionStatus === "connecting");
-
-      if (shouldKillSession) {
-        await walletConnector.killSession();
-        dispatch({ type: "status", status: "disconnected" });
+      setStatus("loading");
+      if (wallet.connected) {
+        await wallet.disconnect();
       }
 
-      dispatch({ type: "status", status: "connecting" });
-
-      const connectionResponse = await walletConnector.connect();
-      // walletConnector.connected has a race condition and may return false even if connected
-      const validResponse = Boolean(connectionResponse?.accounts);
-
-      if (validResponse) {
-        const connectionIsFalsy = !walletConnector.connected;
-
-        if (connectionIsFalsy) {
-          // Does not retrigger wallet connect modal, only updates current connection
-          await walletConnector.connect();
-        }
-
-        const [newAddress] = connectionResponse.accounts;
+      const res = await wallet.connect();
 
-        const isNewAddress = checkNewAddress(newAddress);
+      if (res) {
+        const nonce = await fetchNonce(res.address);
+        const message = process.env.NEXT_PUBLIC_SIGNING_MESSAGE + " " + nonce;
+        const signature = await wallet.signMessageAsync({ message });
+        if (signature) {
+          const addedWallet = await addWalletToBackend({
+            address: res.address,
+            signature,
+          });
 
-        if (isNewAddress) {
-          await addWalletToBackend(newAddress);
-          await disconnectMagic();
+          mutate(MY_INFO_ENDPOINT);
 
-          toastMessage = "Address added and will soon appear on your profile";
-        } else {
-          toastMessage = "Address already connected to your profile";
+          // automatically set the primary wallet on add wallet if user doesn't have one
+          if (hasNoPrimaryWallet) {
+            setPrimaryWallet(addedWallet);
+          }
```

Comment:
💯

Reply:
Jorge is adding it on [backend](https://github.com/showtime-xyz/stbackend/pull/495) so I guess we can remove this!

Reply:
i'm adding the primary wallet if they are a magic user or if they only have the one wallet added! we should probably still prompt them to add one otherwise, i'm going to have the web3 endpoints return an error code for this!

Reply:
> we should probably still prompt them to add one otherwise

yeah definitely! if `primary_wallet` is `null`, they'd see the prompt regardless!

> i'm going to have the web3 endpoints return an error code for this!

Perfect, we'd not let them perform the action if `primary_wallet` is `null`. But good to have backend error as well!